Geena - geena-panzitta/glp_week10assignment.do

```diff
@@ -0,0 +1,267 @@
+
+}
+
+preserve
+drop if enum_1 == 0 | enum_2 == 0 //drop pairs that have a missing enumerator
+collapse enum_1, by(id1) //keep 1 obs for each household
+rename (id1 enum_1)(id enum) //rename to match original data
+save `assignments', replace //save the tempfile
+restore
+
+drop if enum_1 == 0 | enum_2 == 0 //same as above
+collapse enum_2, by(id2)
+rename (id2 enum_2)(id enum)
+append using `assignments' //add to enum_1 assignments
+collapse enum, by(id) //keep 1 obs for each household
+save `assignments', replace //save as tempfile
+
+clear
+use "${gps_data}" //load original data
+merge 1:1 id using `assignments' //merge in enumerator assignment
+sort enum //sort by enumerator
+tab enum //show how many households are assigned to each enum
+
+//ssc inst sepscatter
+sepscatter longitude latitude, sep(enum) //create scatter plot by enumerator
+
+/* creates the scatter plot without sepscatter
+twoway (scatter longitude latitude if enum == 1, m(circle) mc(blue) legend(label(1 "Enumerator 1") label(2 "Enumerator 2") label(3 "Enumerator 3") label(4 "Enumerator 4") label(5 "Enumerator 5") label(6 "Enumerator 6") label(7 "Enumerator 7") label(8 "Enumerator 8") label(9 "Enumerator 9") label(10 "Enumerator 10") label(11 "Enumerator 11") label(12 "Enumerator 12") label(13 "Enumerator 13") label(14 "Enumerator 14") label(15 "Enumerator 15") label(16 "Enumerator 16") label(17 "Enumerator 17") label(18 "Enumerator 18") label(19 "Enumerator 19"))) (scatter longitude latitude if enum == 2, m(circle) mc(red)) (scatter longitude latitude if enum == 3, m(circle) mc(orange)) (scatter longitude latitude if enum == 4, m(circle) mc(pink)) (scatter longitude latitude if enum == 5, m(circle) mc(gray)) (scatter longitude latitude if enum == 6, m(circle) mc(cyan)) (scatter longitude latitude if enum == 7, m(circle) mc(green)) (scatter longitude latitude if enum == 8, m(circle) mc(brown)) (scatter longitude latitude if enum == 9, m(circle) mc(purple)) (scatter longitude latitude if enum == 10, m(circle) mc(black)) (scatter longitude latitude if enum == 11, m(circle) mc(lavender)) (scatter longitude latitude if enum == 12, m(circle) mc(lime)) (scatter longitude latitude if enum == 13, m(circle) mc(yellow)) (scatter longitude latitude if enum == 14, m(circle) mc(teal)) (scatter longitude latitude if enum == 15, m(circle) mc(maroon)) (scatter longitude latitude if enum == 16, m(circle) mc(mint)) (scatter longitude latitude if enum == 17, m(circle) mc(khaki)) (scatter longitude latitude if enum == 18, m(circle) mc(magenta)) (scatter longitude latitude if enum == 19, m(circle) mc(olive))
+*/
+
+}
+
+
+/*******************************************************************************
+Q4
+*******************************************************************************/
+
+{
+
+clear
+use "${tz_elec_10_clean}" //import 2010 data
+rename ward_10 ward //rename variable to match
+sort ward
+by ward, sort: generate i = _n
+drop if i != 1 //keep only one obs for each ward
```

Comment:
Lines 239 through 241 aren't necessary, as some wards might have the same name but are located in different regions. You don't want to drop these wards from the dataset!

Reply:
Same thing with lines 248-250